### ISSUE TITLE
Documentation improvements

### ISSUE THREAD

**Issue by einhverfr** [wrouesnel/postgres_exporter#95]
Hi;

As a note, your documentation as to how to run as a non-superuser is needlessly complex.  In PostgreSQL, views are already security definer and have been for a long time, so no need for the view -> function -> table approach.  Just build a view around the table and a security definer function around the function you need to expose.  I am likely to submit a PR to this effect in the near future.

**Comment by einhverfr** [comment]
(Yes, I have tested this).

**Comment by einhverfr** [comment]
In particular I have tested https://github.com/adjust/pg-telemetry with access by non-superusers to the statement stats views.  As a note, we include a queries.yaml for use with your exporter.

**Comment by sathieu** [comment]
Can you provide SQL statements?

**Comment by einhverfr** [comment]
Instead of:
```
CREATE USER postgres_exporter PASSWORD 'password';
ALTER USER postgres_exporter SET SEARCH_PATH TO postgres_exporter,pg_catalog;

-- If deploying as non-superuser (for example in AWS RDS)
-- GRANT postgres_exporter TO :MASTER_USER;
CREATE SCHEMA postgres_exporter AUTHORIZATION postgres_exporter;

CREATE FUNCTION postgres_exporter.f_select_pg_stat_activity()
RETURNS setof pg_catalog.pg_stat_activity
LANGUAGE sql
SECURITY DEFINER
AS $$
  SELECT * from pg_catalog.pg_stat_activity;
$$;

CREATE FUNCTION postgres_exporter.f_select_pg_stat_replication()
RETURNS setof pg_catalog.pg_stat_replication
LANGUAGE sql
SECURITY DEFINER
AS $$
  SELECT * from pg_catalog.pg_stat_replication;
$$;

CREATE VIEW postgres_exporter.pg_stat_replication
AS
  SELECT * FROM postgres_exporter.f_select_pg_stat_replication();

CREATE VIEW postgres_exporter.pg_stat_activity
AS
  SELECT * FROM postgres_exporter.f_select_pg_stat_activity();

GRANT SELECT ON postgres_exporter.pg_stat_replication TO postgres_exporter;
GRANT SELECT ON postgres_exporter.pg_stat_activity TO postgres_exporter;
```
```
CREATE USER postgres_exporter PASSWORD 'password';
ALTER USER postgres_exporter SET SEARCH_PATH TO postgres_exporter,pg_catalog;

-- If deploying as non-superuser (for example in AWS RDS)
-- GRANT postgres_exporter TO :MASTER_USER;
CREATE SCHEMA postgres_exporter AUTHORIZATION postgres_exporter;

-- views always run as the definer and you assign permissions separately on them
CREATE VIEW postgres_exporter.pg_stat_activity AS
select * from pg_catalog.pg_stat_activity;

CREATE VIEW postgres_exporter.pg_stat_replication AS
SELECT * FROM pg_catalog.pg_stat_replication;

GRANT SELECT ON postgres_exporter.pg_stat_replication TO postgres_exporter;
GRANT SELECT ON postgres_exporter.pg_stat_activity TO postgres_exporter;
```

Much simpler.

**Comment by sathieu** [comment]
@einhverfr great! Van you sens a PR?

**Comment by einhverfr** [comment]
Sure will do a PR :-)

**Comment by Globegitter** [comment]
@wrouesnel Also on the topic of documentation I am just getting started with prometheus and grafana on kubernetes and really not sure how to get this up and running. I have set-up the docker container fine and that is connection to my rds database, and added the right kubernetes annotations so prometheus is aware of it. But now I am kind of stuck. Do I have to add something to the prometheus conf? Or should it just work?

**Comment by Globegitter** [comment]
Ahh I think I figured it out - it is now at least showing up as a target in prometheus - so that is great :)

**Comment by cosmopetrich** [comment]
It seems like this issue can be closed? The change was merged in #103 and is now being discussed again in #134.

**Comment by knweiss** [comment]
@einhverfr Is there a minimum psql version for your simpler version? As mentioned in #217 the simpler version does not work for me in psql 9.3 (on CentOS 7).

#134 seems to indicate that psql <= 9.6 requires the old verison (which I have yet to try).

Maybe the README.md could be updated to mention the required version numbers?
